components/scream/src/share/io/scorpio_input.hpp

```diff
@@ -109,10 +118,12 @@ class AtmosphereInput
 
 protected:
   // Internal functions
-  void register_dimensions(const std::string name);
   void register_variables();
   void set_degrees_of_freedom();
   void register_views();
+  std::vector<std::string> get_vec_of_dims(const std::vector<FieldTag> tags);
+  std::string get_io_decomp(std::vector<std::string> vec_of_dims);
```

Comment:
Albeit not really important, as a C++ programmer it hurts to see std::vector's and std::string' passed by value instead of by reference...

Reply:
should I pass all `std::` objects by reference?

Reply:
Usually, yes.